### DISCUSSION TITLE
CHANGING NAME OF PREDICTED LABEL ON PREDICTED IMAGE

### DISCUSSION THREAD

**Discussion by edanurkamaci** [ultralytics/ultralytics#2254] [Q&A]
Hello i have an urgent question. I do not have time to train a model again so i wanted to change the name of labels after test result, but i want new labels on predicted result image. How can i do that?

**Comment by dh031200** [comment]
@edanurkamaci 

You can change the name using the code below.
This is the `person` -> `human` example.

```python
import torch
from ultralytics import YOLO

model = YOLO('yolov8n.pt')

print('Before: ')
print(model.model.names)
print()


indices = list(model.model.names.values())

# Change name here
## model.model.names[indices.index(BEFORE)] = AFTER
model.model.names[indices.index('person')] = 'human'

print('After: ')
print(model.model.names)


ckpt = {
    'epoch': model.ckpt['epoch'],
    'best_fitness': model.ckpt['best_fitness'],
    'model': model.ckpt['model'],
    'ema': model.ckpt['ema'],
    'updates': model.ckpt['updates'],
    'optimizer': model.ckpt['optimizer'],
    'train_args': model.ckpt['train_args'],
    'date': model.ckpt['date'],
    'version': model.ckpt['version']}

torch.save(ckpt, 'rename.pt')
```
## Result
![image](https://user-images.githubusercontent.com/66017052/235595271-c974dc18-43af-41f1-ba0e-853a7da15ddd.png)


Before:  
![bus](https://user-images.githubusercontent.com/66017052/235594303-3cf6452d-fd0f-4270-a580-4495e6ce5e99.jpg)

After:  
![bus](https://user-images.githubusercontent.com/66017052/235594348-c3cdb919-932e-440b-837e-48458fb4a117.jpg)

**Reply by edanurkamaci** [reply]
Thanks a lot. It worked

**Reply by glenn-jocher** [reply]
@edanurkamaci i'm glad to hear that it worked for you! If you have any more questions or need further assistance, feel free to reach out. Happy detecting! 😊🚀
